react/tic-tac-toe/src/redux/login/actions.js

```diff
@@ -0,0 +1,30 @@
+import dataFetch from '../../services/MatchesService';
+
+export const actions = {
+  SET_AUTH: '@@DATA/SET_AUTH',
+  LOGIN_REQUEST: '@@DATA/LOGIN_REQUEST',
+  LOGIN_SUCCESS: '@@DATA/GLOGIN_SUCCESS',
+  LOGIN_FAILURE: '@@DATA/LOGIN_FAILURE'
+};
+
+const actionsCreators = {
+  onLogin: values => async dispatch => {
+    dispatch({ type: actions.LOGIN_REQUEST });
+    try {
+      const res = await dataFetch.onLogin(values);
+      if (res.status === 200) {
+        localStorage.setItem('token', res.data.token);
+        return dispatch({ type: actions.SET_AUTH, hasToken: true, isError: false });
+      }
+      return dispatch({ type: actions.SET_AUTH, hasToken: false, isError: true });
+    } catch (error) {
+      return error;
```

Comment:
dispatch the error here too instead of returning the error, 
you don't read the return of this function anywhere (which is ok)